packages/git/src/browser/diff/git-diff-service.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { GitLogOptions } from './git-diff-model';
+import { injectable, inject } from "inversify";
+import { GitRepositoryProvider } from '../git-repository-provider';
+import { Git, Repository, GitFileChange, GitUtils } from '../../common';
+import { FileUri } from '@theia/core/lib/node/file-uri';
+import * as Path from 'path';
+
+@injectable()
+export class GitDiffService {
+
+    constructor(
+        @inject(Git) protected readonly git: Git,
+        @inject(GitRepositoryProvider) protected readonly repositoryProvider: GitRepositoryProvider
+    ) { }
+
+    async getDiff(repository: Repository, options: GitLogOptions): Promise<GitFileChange[]> {
```

Comment:
+1 for renaming to `log` and moving it to `git.ts` with the `GitLogOptions`.

Reply:
Ok, but should be `diff` then.

Reply:
Under the hood, it runs `git log` right? Also, you have the `log options` argument, why would you call it to `diff`?

Reply:
You are absolutely right, it is `diff`. It resolves to an array of file changes, instead of commit fragments. Please ignore my comments.

Reply:
See #1113.